x-pack/plugins/security_solution/public/endpoint_alerts/store/middleware.ts

```diff
@@ -13,14 +13,34 @@ import {
 import { AlertConstants } from '../../../common/endpoint_alerts/alert_constants';
 import { ImmutableMiddlewareFactory } from '../../common/store';
 import { cloneHttpFetchQuery } from '../../common/utils/clone_http_fetch_query';
+
 import {
   isOnAlertPage,
   apiQueryParams,
-  hasSelectedAlert,
   uiQueryParams,
+  hasSelectedAlert,
   isAlertPageTabChange,
 } from './selectors';
 
+let lastSelectedAlert: string | null = null;
+/**
+ * @returns <boolean> true once per change of `selectedAlert` in query params.
+ *
+ * As opposed to `hasSelectedAlert` which always returns true if the alert is present
+ * query params, which can cause unnecessary requests and re-renders in some cases.
+ */
+const selectedAlertHasChanged = (params: ReturnType<typeof uiQueryParams>): boolean => {
+  const { selected_alert: selectedAlert } = params;
+  if (typeof selectedAlert !== 'string') {
```

Comment:
Nice one here. Query params can be a pain to deal with with re-renders. Kinda following on my earlier if statement grouping
```
const shouldNotChange = typeof selectedAlert !== 'string' || selectedAlert === lastSelectedAlert'

if (shouldNotChange) return false;
lastSelectedAlert = selectedAlert;
return true;
```

Reply:
TS _really_ wanted that `typeof` check on its own, but I moved it closer to what you're suggesting here. Thank! https://github.com/elastic/kibana/pull/69098/commits/96732aef84aeb5b5d3320680bf312abbbbb44e21